src/ts/components/settingspanel.ts

```diff
@@ -93,78 +121,157 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
     this.updateActivePageClass();
   }
 
-  private updateActivePageClass(): void {
-    this.getPages().forEach((page: SettingsPanelPage, index) => {
-      if (index === this.activePageIndex) {
-        page.getDomElement().addClass(this.prefixCss(SettingsPanel.CLASS_ACTIVE_PAGE));
-      } else {
-        page.getDomElement().removeClass(this.prefixCss(SettingsPanel.CLASS_ACTIVE_PAGE));
-      }
-    });
+  /**
+   * Returns the current active / visible page
+   * @return {SettingsPanelPage}
+   */
+  getActivePage(): SettingsPanelPage {
+    return this.activePage;
   }
 
+  /**
+   * Sets the
+   * @deprecated Use setActivePage instead
+   * @param index
+   */
   setActivePageIndex(index: number): void {
-    const targetPage = this.getPages()[index];
-    if (targetPage) {
-      this.animateNavigation(targetPage);
-      this.activePageIndex = index;
-      this.navigationStack.push(targetPage);
-      this.updateActivePageClass();
-      targetPage.onActiveEvent();
-    }
+    this.setActivePage(this.getPages()[index]);
   }
 
-  setActivePage(page: SettingsPanelPage): void {
-    const index = this.getPages().indexOf(page);
-    this.setActivePageIndex(index);
+  /**
+   * Adds the passed page to the navigation stack and make it visible. Do not use this method to navigate backwards!
+   * Results in no-op if the target pate is the current page.
+   * @params page
+   */
+  setActivePage(targetPage: SettingsPanelPage): void {
+    if (targetPage === this.getActivePage()) {
+      console.warn('Page is already the current one ... skipping navigation');
+      return;
+    }
+
+    this.navigateToPage(targetPage, this.getActivePage(), NavigationDirection.Forward);
   }
 
+  /**
+   * Resets the navigation stack and navigate to the root page and make it visible
```

Comment:
"Resets the navigtion stack by navigating back to the root page and displaying it."